cmd/validate.go

```diff
@@ -66,6 +84,205 @@ this command.
 	},
 }
 
+func validateEntities(ctx context.Context, obj interface{}, kongClient *kong.Client, entityType string) []error {
+	entities, err := callGetAll(obj)
+	if err != nil {
+		return []error{err}
+	}
+	errors := []error{}
+
+	// create a buffer of channels. Creation of new coroutines
+	// are allowed only if the buffer is not full.
+	chanBuff := make(chan struct{}, maxConcurrency)
+
+	var wg sync.WaitGroup
+	wg.Add(entities.Len())
+	// each coroutine will append on a slice of errors.
+	// since slices are not thread-safe, let's add a mutex
+	// to handle access to the slice.
+	mu := &sync.Mutex{}
+	for i := 0; i < entities.Len(); i++ {
+		// reserve a slot
+		chanBuff <- struct{}{}
+		go func(i int) {
+			defer wg.Done()
+			// release a slot when completed
+			defer func() { <-chanBuff }()
+			_, err := validateEntity(ctx, kongClient, entityType, entities.Index(i).Interface())
+			if err != nil {
+				mu.Lock()
+				errors = append(errors, err)
+				mu.Unlock()
+			}
+		}(i)
+	}
+	wg.Wait()
+	return errors
+}
+
+func validateWithKong(cmd *cobra.Command, ks *state.KongState, targetContent *file.Content) []error {
+	ctx := cmd.Context()
+	// make sure we are able to connect to Kong
+	_, err := fetchKongVersion(ctx, rootConfig)
+	if err != nil {
+		return []error{fmt.Errorf("couldn't fetch Kong version: %w", err)}
+	}
+
+	// check if workspace exists
+	workspaceName := getWorkspaceName(validateWorkspace, targetContent)
+	workspaceExists, err := workspaceExists(ctx, rootConfig, workspaceName)
+	if err != nil {
+		return []error{err}
+	}
+	if !workspaceExists {
```

Comment:
What if decK is used against the open-source version of Kong (which has no concept of workspace)?
All workspace specific code must be executed only when the user supplies a workspace.